watson-developer-cloud/discovery-components
# feat: document provider interface

#### What do these changes do/fix?

Allow consumer to specify an implementation of `DocumentProvider`, which will be used by `DocumentPreview` to query for the original document (PDF) to display. If not provided, continue previewing HTML or text.

#### How do you test/verify these changes?
Visit DocumentPreview in Storybook, for each sub-stories, check if PDF document type preview displays PDF. Note that highlighting is not ready in PDF.

In the Example Discovery search app, open a PDF named `Art Effects.pdf` (or you can choose any name in `ExampleDocumentProvider`), then see if the PDF is rendered correctly as DocumentPreview.

#### Have you documented your changes (if necessary)?
N/A

#### Are there any breaking changes included in this pull request?
No
